### DISCUSSION TITLE
show page status permanently?

### DISCUSSION THREAD

**Discussion by fautsie** [KBLiveSolutions/open.control#23] [Troubleshooting]
Hey Guys,
 
i just received my open control maker a couple of days ago and everything seems to work great! combined with a browser based editor it´s great coding with great features packed in a small format! really love it so far!!! 

but as the title says: isnt there a way to show which page your on like permantly? 
(maybe just thin vertical lines in the display or on a certain colour on a led)

and also for the leds: cant i just send midi notes to them and a certain velcoity selects the colour of it? 
cause how could i "map" an led to an ableton device to show if its on or off? maybe i´ve overseen something

and the mute function for the leds is nice but it just works with the selcted track! it would be really great if you could select a certain track  to show on the led if its muted or not! 

sorry don´t wanna moan but i have so many ideas with this pedal! its really really great!
i just need to really customize it for my needs! 

thanks for your help in advance!

**Comment by KBLiveSolutions** [comment]
> Hey Guys,
> 
> i just received my open control maker a couple of days ago and everything seems to work great! combined with a browser based editor it´s great coding with great features packed in a small format! really love it so far!!!
> 

thank you ! 

> but as the title says: isnt there a way to show which page your on like permantly? (maybe just thin vertical lines in the display or on a certain colour on a led)
> 

Yes, it's a good idea ; I imagine the first LED column in the display could show 1, 2 or 3 dots. 
I could also add a "Page color" entry for the RGB LEDs.
I'll try something when I have some time (not sure how soon it will be).

> and also for the leds: cant i just send midi notes to them and a certain velcoity selects the colour of it? cause how could i "map" an led to an ableton device to show if its on or off? maybe i´ve overseen something
> 
> and the mute function for the leds is nice but it just works with the selcted track! it would be really great if you could select a certain track to show on the led if its muted or not!
> 

I think both requests could be solved by adding a  "Custom MIDI" entry for the RGB LEDs (ie you have a button with Custom MIDI, and choose the same CC for the LED).
Again, I'll see what I can do when I find a moment.

> sorry don´t wanna moan but i have so many ideas with this pedal! its really really great! i just need to really customize it for my needs!
> 

No problem at all, open·control is still under development, and all Feature Requests are welcome if they make sense !

> thanks for your help in advance!

May I ask you to show what you've done with your Maker board when you have something to share ?

**Comment by fautsie** [comment]
Hello Pierre,

first off all i have to say even now some things are not working out the way i want them but i am really happy with your open control maker and i think it´s an absolute great price value!! 


> Yes, it's a good idea ; I imagine the first LED column in the display could show 1, 2 or 3 dots.
    I could also add a "Page color" entry for the RGB LEDs.
    I'll try something when I have some time (not sure how soon it will be).

either way would work for me! it´s just that i can see permanently which page i´m on atm...  that would be great if you can find time for that...


> I think both requests could be solved by adding a "Custom MIDI" entry for the RGB LEDs (ie you have a button with Custom MIDI, 
    and choose the same CC for the LED).
    Again, I'll see what I can do when I find a moment.

thats exactly what i was thinking. if the LED´s would have an CUSTOM MIDI feature that would solve lot´s of my requests


> No problem at all, open·control is still under development, and all Feature Requests are welcome if they make sense !

great attitude!! 


> May I ask you to show what you've done with your Maker board when you have something to share ?

yes, for sure!!! i didn´t even think of doing a "making of" video but i do understand now that it could help imagining how easy it is to put your own board together so as a little reward for helping me out with coding i definately will make a video about this footswitch.
and i have already a very cool idea of making the housing.... together with my son.... 
_ _ _ _ 


Ok, but i wanna keep you updated how it´s going so far and i think its quite important for you to know what i´m doing with it...

i make my income with ableton live looping (I´m a multi Instrumentalist) since nearly nine years so i´m doing this professionally and i play 70-80 gig a year.
thats why my setup (hardware and software)  got bigger and bigger (ableton Set with nearly 60 Tracks and 20 ins + 20 outs through a RME Interface)  and it´s all very customized.

i do use touch osc a lot (mainly when i play with my looping band) for all external devices (ipads iphones etc) 
but i did make custom midi remote scripts with Control Surface Studio for my NI Jam Controller and it works really reliable! 
and the most important factor is that i can rely on my setup! 

So unfortunately now that i have everything whats important to me set up with the open control i get sort of an error... meaning that as soon i have selected the open control script in ableton it seems it clashes with my other remote scripts...  and it cant update any changes from the editor anymore... and doesnt read out the looper status anymore for the display (and i do the same thing with NI Jam Script)
but i tried  it on an empty set and it was working so it´s definately a bug beetween the scripts...

but atm thats not a really big issue for me cause i plan to use the board from autumn ish for the upcoming winter season so i have plenty time to sort that out...

the footswitch that i´m using since 3-4 years is a diy project with a teensy 3.2 board (PJRC) in it but it´s very basic due to i´m not a coding pro!  but i can imagine that your open control board will really fulfill my dreams!!! so i´m very much looking forward to the process of customizing it to my needs even though it conflicts with my scripts atm ...

have a really good time and thank you so much for helping me out a bit!!

greeting from austria!

**Reply by KBLiveSolutions** [reply]
Hello

I have updated the script with a few new features :
- there is now a "Custom MIDI" field for LEDs in the Editor. Right now the color is fixed for each LED
- there is a new "Page Color" field for the LEDs in the Editor. Same, you can't choose the color ATM.
- The display shows 1, 2 or 3 dots to reflect the current page number

To upgrade, you need to :
- [download](https://github.com/KBLiveSolutions/open.control-firmware/blob/webusb-support/open_control_firmware_1.5.ino.uf2) and install a new version of the firmware (instructions [here](https://github.com/KBLiveSolutions/open.control-firmware))
- [download](https://github.com/KBLiveSolutions/open.control-remote_script/archive/refs/heads/custom-actions.zip) and install a new version of the Remote Script
- [download](https://github.com/KBLiveSolutions/open.control-editor/archive/refs/heads/webusb.zip) a new version of the Editor, and use the index.html inside it.

Also, this update is part of a larger update that is still in beta testing and that includes (among other features) a new connection method for the editor. More details [here](https://github.com/KBLiveSolutions/open.control/discussions/12).

Please let me know if everything works ok or report any misbehavior.
Cheers
p
